src/ucp/api/ucp.h

```diff
@@ -402,6 +414,14 @@ typedef struct ucp_params {
      * This field defaults to 0 if not specified.
      */
     uint64_t                           tag_sender_mask;
+
+    /**
+     * This flag indicates if this context is shared by multiple workers
+     * from different threads. If so, this context needs thread safety
```

Comment:
need to clarify this more - give several examples when MT is needed and when is not